Update whitelist.txt

egrace479 opened the pull request:
Preliminary whitelist with all container contents and baseline submission requirements.

egrace479 committed:
Update whitelist.txt

preliminary whitelist with all container contents and baseline submission requirements

hlapp commented on gw_challenge/ingested_program/whitelist.txt:
```diff
@@ -7,3 +7,14 @@ torch
+Pillow
+scipy
+tqdm
+pyarrow
```
Should perhaps include duckdb if we're including this?

egrace479 replied:
```suggestion
pyarrow
duckdb
```
Seems reasonable.

ytchoutw commented on gw_challenge/ingested_program/whitelist.txt:
```diff
@@ -7,3 +7,14 @@ torch
+nflows
+lightgbm
+seaborn
+iminuit
```
also include the following
numpy
scikit-learn
torchaudio
Keras
catboost
plotly

work4cs replied:
```suggestion
iminuit
timm
torchaudio
catboost
plotly
```

work4cs replied:
@ytchoutw Thank you for adding more packages! I use line commit to add some of them. For numpy, scikit-learn, Keras, they are before Line 7. Unfortunately, it is not very convenient to view the full file.

egrace479 replied:
For future reference, three dots at the top right (next to comment and view) have the option to "view file" and clicking on the blue bar in the file preview in this window will expand to show more since GitHub truncates to focus on the part that has been changed.

egrace479 committed:
Update ingestion.py

expand versioning options for whitelist

egrace479 commented:
I added the recommended [version regex from PyPI](https://packaging.python.org/en/latest/specifications/version-specifiers/#version-specifiers) (it's more expansive) as suggested by @work4cs for the whitelist test as well.

egrace479 committed:
Apply suggestions from code review

update the whitelist to match change on imageomics

https://github.com/Imageomics/HDR-anomaly-challenge/blob/2e54051b587d0a890c426512fa3bd177b8b0a753/ingestion_program/whitelist.txt

Co-authored-by: Lisa <47147576+work4cs@users.noreply.github.com>

ytchoutw commented on gw_challenge/ingested_program/whitelist.txt:
```diff
@@ -7,3 +7,19 @@ torch
+timm
+torchaudio
+catboost
+plotly
```
```suggestion
plotly
netCDF4
xarray
```

ytchoutw reviewed (commented):
Add packages for iHarp

ytchoutw commented:
Shall we merge this already and update it to Codabench? I think we have most of the required packages

egrace479 committed:
add iHarp packages

Co-authored-by: Yuan-Tang Chou <ytchou97@gmail.com>

egrace479 commented:
We can always add more later, as we did set it up to tell participants to request unavailable packages through GitHub.

katyagovorkova commented on gw_challenge/ingested_program/ingestion.py:
```diff
@@ -8,7 +8,7 @@
 
 # expected version pattern for requirements
-VERSION_PATTERN = re.compile("^[0-9].[0-9].[0-9]$")
+VERSION_PATTERN = re.compile("^[N!]N(.N)*[{a|b|rc}N][.postN][.devN]$")
```
sorry I only saw it after merging, but upon testing, this version pattern doesn't seem to be doing what we want? For example, version "2.15.0" does not match the pattern, is it expected behaviour? Thanks 

@egrace479 @work4cs

egrace479 replied:
Yeah, that does not seem to be working. @work4cs, this was your update on ours, could you please resolve this or revert both back to the pattern I initially set up? Thanks!

work4cs replied:
I may not revert to the old version. A period mean anything. So, "1@2@3" will also match. Instead, I will convert this `^[N!]N(.N)*[{a|b|rc}N][.postN][.devN]$` into python regex verion.

work4cs replied:
We can use [`package`](https://packaging.pypa.io/en/latest/version.html)

`from packaging.version import Version
Version('1.2@3') #raise an exception
Version('1.2.3') #return an object
` 
`if not Version(version)` will be enough

hlapp replied:
I agree, that seems like the right thing to use rather than come up with our custom reinvention. Nice find @work4cs.

katyagovorkova replied:
@work4cs interesting solution! Could you please add it to the code and make another pull request? Thanks!

work4cs replied:
I updated [butterfly challenge](https://github.com/Imageomics/HDR-anomaly-challenge/pull/30). Not sure how to update here. Which repo should I update, egrace479:HDRchallenge or a3d3-institute:HDRchallenge?

katyagovorkova replied:
thanks for pointing out to the changes, I will transfer them to this repo!

work4cs replied:
I wrote down and tested two ways to handle the situation of the version does not match the regex:

1. throw exception and exit (currently using)
2. catch the exception of error, print the error message and install the default version (commented out)
I currently use 1 because the 2's message is easily ignored among all the installing messages in the console. And if the default version we automatically install for them does not work, it might be hard for the participants to figure out we install another version and it causes some errors.

egrace479 replied:
Sounds like a good plan @work4cs, thanks!
